### PR TITLE
#38: ignore decoration nodes in findDomRefAtPos

### PR DESCRIPTION
**Pull request by eshvedai**
see #38

### CODE REVIEW

File: src/selection.js [hunk 1]
```diff
@@ -164,6 +164,17 @@ export const findPositionOfNodeBefore = selection => {
 // ```
 export const findDomRefAtPos = (position, domAtPos) => {
   const dom = domAtPos(position);
-  const node = dom.node.childNodes[dom.offset];
+  let { offset } = dom;
+  for (let i = 0, count = dom.node.childNodes.length; i < count; i++) {
+    const child = dom.node.childNodes[i];
+    // ignore decoration DOM nodes as ProseMirror ignores it while calculating the offset
+    if (
+      child.nodeType === 1 &&
```
Comment (5id) [inline src/selection.js]: `child.nodeType === Node.ELEMENT_NODE`

File: src/selection.js [hunk 2]
```diff
@@ -164,6 +164,17 @@ export const findPositionOfNodeBefore = selection => {
 // ```
 export const findDomRefAtPos = (position, domAtPos) => {
   const dom = domAtPos(position);
-  const node = dom.node.childNodes[dom.offset];
+  let { offset } = dom;
+  for (let i = 0, count = dom.node.childNodes.length; i < count; i++) {
```
Comment (5id) [inline src/selection.js]: If I have something like:
```
let { offset, node } = dom;
offset = 1;
node.childNodes = [decoration, decoration, myThing, decoration, decoration];
```
won't this set the offset to `5`?

Comment (eshvedai) [reply src/selection.js]: I'll add a test for it

Comment (eshvedai) [reply src/selection.js]: all done

File: __tests__/selection.js [hunk 3]
```diff
@@ -361,5 +361,38 @@ describe('selection', () => {
...
+      const deco1 = document.createElement('div');
+      deco1.className = 'fake-decoration1 ProseMirror-widget';
+      view.dom.insertBefore(deco1, view.dom.firstChild);
+
+      // prepend a fake decoration node 1
+      const deco2 = document.createElement('div');
+      deco2.className = 'fake-decoration2 ProseMirror-widget';
+      view.dom.insertBefore(deco2, view.dom.childNodes[1]);
+
+      // prepend a fake decoration node 1
```
Comment (ahixon) [inline __tests__/selection.js]: append*

Comment (eshvedai) [reply __tests__/selection.js]: 👍

**Pull request merged**